service/history/hsm/tree.go

```diff
@@ -601,3 +703,23 @@ func GenerateEventLoadToken(event *historypb.HistoryEvent) ([]byte, error) {
 	}
 	return proto.Marshal(ref)
 }
+
+func (n *Node) getRoot() *Node {
```

Comment:
https://go.dev/doc/effective_go#Getters

```suggestion
func (n *Node) root() *Node {
```